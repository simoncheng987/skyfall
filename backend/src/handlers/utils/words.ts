import { Word } from '../../types';

const roomInfo = {};

export const getWordForRoom = (roomId: string, words: Array<string>): Word => {
  // if word was never emitted, initialise id to 0
  if (!roomInfo[roomId]) {
    roomInfo[roomId] = {
      wordId: 0,
    };
  }

  // get random word, create Word object
  const randomWord: Word = {
    word: words[Math.floor(Math.random() * words.length)],
    id: String(roomInfo[roomId].wordId),
  };

  roomInfo[roomId].wordId++;

  return randomWord;
};
