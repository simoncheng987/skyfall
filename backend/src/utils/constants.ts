import 'dotenv/config';

export const port = process.env.PORT || '...You have not supplied PORT number!';
export const mongodbUri = process.env.MONGODB || '...You have not supplied MONGODB url!';
export const MAX_PLAYERS = 2;
