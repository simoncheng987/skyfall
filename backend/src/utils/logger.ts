/* eslint-disable */

const info = (...msg: any) => console.log(...msg)
const error = (...msg: any) => console.error(...msg)

const logger = {
    info,
    error
}
export default logger