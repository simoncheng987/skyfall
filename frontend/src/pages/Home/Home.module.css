.columnFlex {
  display: flex;
  flex-direction: column;
}

.buttons {
  margin-top: 5px;
}

.buttons > *:not(:first-child) {
  margin-top: 20px;
}

.button {
  display: block;
  width: 100%;
  height: 70px;
}

.leaderboard {
  text-align: right;
  height: 3vw;
  color: #ffffff;
  position: absolute;
  top: 60px;
  right: 60px;
  transition: 0.2s all ease-in-out;
  cursor: pointer;
}

.leaderboard:hover {
  opacity: 0.5;
}

.leaderboard:hover {
  opacity: 0.7;
}

.container {
  display: flex;
  flex-direction: row;
  width: 100vw;
  justify-content: space-evenly;
}
