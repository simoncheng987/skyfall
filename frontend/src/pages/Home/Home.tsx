import React from 'react';
import { useNavigate } from 'react-router-dom';
import { ChartBarIcon } from '@heroicons/react/outline';
import Button from '../../components/Button';
import PageScaffold from '../PageScaffold';
import Title from '../../components/Title';
import styles from './Home.module.css';

export default function Home() {
  const navigate = useNavigate();

  return (
    <PageScaffold>
      <Title text="Skyfall" colorScheme="yellow" fontSize="130px" />
      <div className={styles.buttons}>
        <Button
          text="Create Room"
          className={styles.button}
          buttonStyle="secondary"
          onClick={() => navigate('create')}
          ariaLabel="create-room-button"
        />
        <Button
          text="Join Room"
          className={styles.button}
          onClick={() => navigate('join')}
          ariaLabel="join-room-button"
        />
      </div>
      <ChartBarIcon className={styles.leaderboard} onClick={() => navigate('leaderboard')} />
    </PageScaffold>
  );
}
