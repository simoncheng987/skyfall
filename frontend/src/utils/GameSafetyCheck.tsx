import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useClient } from '../context/ClientProvider';

export default function GameSafetyCheck() {
  const navigate = useNavigate();
  const { client, clearContext } = useClient();

  useEffect(() => {
    if (!client) {
      clearContext();
      navigate('/');
    }
  }, [client]);
}
