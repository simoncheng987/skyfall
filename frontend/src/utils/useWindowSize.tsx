import { useEffect, useState } from 'react';

interface WindowSize {
  width: number;
  height: number;
}

export default function useWindowSize(): WindowSize {
  if (window === undefined) {
    return {
      width: 1200,
      height: 800,
    };
  }

  const [windowSize, setWindowSize] = useState({
    width: window.innerWidth,
    height: window.innerHeight,
  });

  function changeWindowSize() {
    setWindowSize({ width: window.innerWidth, height: window.innerHeight });
  }

  useEffect(() => {
    window.addEventListener('resize', changeWindowSize);

    return () => {
      window.removeEventListener('resize', changeWindowSize);
    };
  }, []);

  return windowSize;
}
