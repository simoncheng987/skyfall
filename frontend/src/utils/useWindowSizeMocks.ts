import useWindowSize from './useWindowSize';

const mockedHook = useWindowSize as jest.MockedFunction<typeof useWindowSize>;

export function mockLargeWindowSize() {
  mockedHook.mockReturnValue({
    width: 1920,
    height: 1080,
  });
}

export function mockSmallWindowSize() {
  mockedHook.mockReturnValue({
    width: 1000,
    height: 562.5,
  });
}
