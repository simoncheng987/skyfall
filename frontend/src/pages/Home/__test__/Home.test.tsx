/* eslint-disable testing-library/render-result-naming-convention */

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { MemoryHistory, createMemoryHistory } from 'history';
import Home from '../Home';
import { mockLargeWindowSize, mockSmallWindowSize } from '../../../utils/useWindowSizeMocks';

jest.mock('../../../utils/useWindowSize');

function componentWithRouterFactory(memoryHistory: MemoryHistory) {
  return (
    <Router location={memoryHistory.location} navigator={memoryHistory}>
      <Home />
    </Router>
  );
}

function renderComponentWithRouter() {
  const memoryHistory = createMemoryHistory({ initialEntries: ['/'] });
  render(componentWithRouterFactory(memoryHistory));
  return memoryHistory;
}

const SMALL_SCREEN_ERROR =
  /Oh no! Ensure that your monitor size is at least 1280 x 720, and display ratio is greater than 16:10/i;

describe('Home', () => {
  test('all component are rendered to the screen', () => {
    mockLargeWindowSize();
    renderComponentWithRouter();
    expect(screen.getByText('Skyfall')).toBeInTheDocument();
    expect(screen.getByLabelText('create-room-button')).toBeInTheDocument();
    expect(screen.getByLabelText('join-room-button')).toBeInTheDocument();
    expect(screen.getByLabelText('upload-words-button')).toBeInTheDocument();
    expect(screen.getByLabelText('leaderboard-icon')).toBeInTheDocument();
  });

  test('integration test: PageScaffold error is rendered in small screens', () => {
    mockSmallWindowSize();
    renderComponentWithRouter();
    expect(screen.getByText(SMALL_SCREEN_ERROR)).toBeInTheDocument();
  });

  it('routing from Home to CreateRoom page', () => {
    mockLargeWindowSize();
    const history = renderComponentWithRouter();
    const createRoomButton = screen.getByLabelText('create-room-button');
    fireEvent.click(createRoomButton);
    expect(history.location.pathname).toEqual('/create');
  });

  it('routing from Home to JoinRoom page', () => {
    mockLargeWindowSize();
    const history = renderComponentWithRouter();
    const joinRoomButton = screen.getByLabelText('join-room-button');
    fireEvent.click(joinRoomButton);
    expect(history.location.pathname).toEqual('/join');
  });

  it('routing from Home to UploadWords page', () => {
    mockLargeWindowSize();
    const history = renderComponentWithRouter();
    const uploadWordsButton = screen.getByLabelText('upload-words-button');
    fireEvent.click(uploadWordsButton);
    expect(history.location.pathname).toEqual('/upload-words');
  });
});
