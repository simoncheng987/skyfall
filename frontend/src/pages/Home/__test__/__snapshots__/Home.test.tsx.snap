// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default test matches snapshot with home page 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": <Home />,
    "initialEntries": Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ],
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "basename": undefined,
      "children": <Home />,
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "navigationType": "POP",
      "navigator": Object {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "index": 0,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {},
      "rendered": Object {
        "instance": null,
        "nodeType": "component",
        "props": Object {
          "children": Array [
            <Title
              className=""
              colorScheme="yellow"
              fontSize="130px"
              text="Skyfall"
            />,
            <div
              className="buttons"
            >
              <Button
                buttonStyle="secondary"
                className="button"
                onClick={[Function]}
                text="Create Room"
              />
              <Button
                buttonStyle="primary"
                className="button"
                onClick={[Function]}
                text="Join Room"
              />
            </div>,
          ],
          "grassEnabled": true,
        },
        "rendered": Object {
          "instance": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="children"
              >
                <Title
                  className=""
                  colorScheme="yellow"
                  fontSize="130px"
                  text="Skyfall"
                />
                <div
                  className="buttons"
                >
                  <Button
                    buttonStyle="secondary"
                    className="button"
                    onClick={[Function]}
                    text="Create Room"
                  />
                  <Button
                    buttonStyle="primary"
                    className="button"
                    onClick={[Function]}
                    text="Join Room"
                  />
                </div>
              </div>,
              <img
                alt="grass background"
                className="grassImage"
                src="/assets/grass.png"
              />,
            ],
            "className": "container",
          },
          "rendered": Array [
            Object {
              "instance": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Title
                    className=""
                    colorScheme="yellow"
                    fontSize="130px"
                    text="Skyfall"
                  />,
                  <div
                    className="buttons"
                  >
                    <Button
                      buttonStyle="secondary"
                      className="button"
                      onClick={[Function]}
                      text="Create Room"
                    />
                    <Button
                      buttonStyle="primary"
                      className="button"
                      onClick={[Function]}
                      text="Join Room"
                    />
                  </div>,
                ],
                "className": "children",
              },
              "rendered": Array [
                Object {
                  "instance": null,
                  "nodeType": "component",
                  "props": Object {
                    "className": "",
                    "colorScheme": "yellow",
                    "fontSize": "130px",
                    "text": "Skyfall",
                  },
                  "rendered": Object {
                    "instance": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Skyfall",
                      "className": " fontStyle  yellow",
                      "style": Object {
                        "fontSize": "130px",
                      },
                    },
                    "rendered": Array [
                      "Skyfall",
                    ],
                    "type": "div",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Button
                        buttonStyle="secondary"
                        className="button"
                        onClick={[Function]}
                        text="Create Room"
                      />,
                      <Button
                        buttonStyle="primary"
                        className="button"
                        onClick={[Function]}
                        text="Join Room"
                      />,
                    ],
                    "className": "buttons",
                  },
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "nodeType": "component",
                      "props": Object {
                        "buttonStyle": "secondary",
                        "className": "button",
                        "onClick": [Function],
                        "text": "Create Room",
                      },
                      "rendered": Object {
                        "instance": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Create Room",
                          "className": "button button text secondary",
                          "onClick": [Function],
                          "type": "button",
                        },
                        "rendered": Array [
                          "Create Room",
                        ],
                        "type": "button",
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "nodeType": "component",
                      "props": Object {
                        "buttonStyle": "primary",
                        "className": "button",
                        "onClick": [Function],
                        "text": "Join Room",
                      },
                      "rendered": Object {
                        "instance": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Join Room",
                          "className": "button button text primary",
                          "onClick": [Function],
                          "type": "button",
                        },
                        "rendered": Array [
                          "Join Room",
                        ],
                        "type": "button",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "nodeType": "host",
              "props": Object {
                "alt": "grass background",
                "className": "grassImage",
                "src": "/assets/grass.png",
              },
              "rendered": Array [],
              "type": "img",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;
